MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1116,7 +1118,8 @@ private MediaSourceInfo GetVersionInfo(bool enablePathSubstitution, BaseItem ite
                 RunTimeTicks = item.RunTimeTicks,
                 Container = item.Container,
                 Size = item.Size,
-                Type = type
+                Type = type,
+                HasSegments = item.MediaType is MediaType.Video && (protocol is null or MediaProtocol.File) && MediaSegmentManager.HasSegments(item.Id)
```

Comment:
media segment could work for audio types too, e.g. podcasts/audiobooks. Perhaps it makes sense to move the MediaType check to MediaSegmentManager so we can manage the "supported types" over there